src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -436,6 +438,16 @@ module.exports = function<T, P, I, TI, C, CX>(
     } else {
       // Proceed under the assumption that this is a functional component
       workInProgress.tag = FunctionalComponent;
+      if (__DEV__) {
+        warning(
+          !!workInProgress.ref,
+          'Stateless function components cannot be given refs ' +
+          '(See ref "' + workInProgress.ref._stringRef + '" in ' +
+          getComponentName(workInProgress) + ' created by ' +
+          getComponentName(workInProgress.return) + '). ' +
```

Comment:
This is wrong I think. `return` will be parent, not owner. For example in `<div><Foo /></div>`, `Foo`'s `return` is `div`.

I think you should use `ReactDebugCurrentFiber.getCurrentFiberOwnerName()` instead. And in fact maybe you can use `ReactDebugCurrentFiber.getCurrentFiberStackAddendum()` to add a nice stack instead of "see ref". You could change this in Stack too by using `ReactComponentTreeHook.getStackAddendumByID(this._debugID)`.